agent/lib/internal_address_source.js

```diff
@@ -193,7 +193,7 @@ AddressSource.prototype.updated = function (objects) {
     if (changes) {
         this.update_readiness(changes);
         this.dispatch('addresses_defined', addresses, changes.description);
-        this.dispatch_if_changed('addresses_ready', objects.filter(ready).map(extract_spec), same_address_definition);
+        this.dispatch_if_changed('addresses_ready', addresses.filter(ready), same_address_definition);
```

Comment:
I'm worried this might cause a regression, where the entries in the addresses array would have their state mutated, causing the readiness to be wrong (I vaguely remember or imagining fixing an issue like that).

Reply:
I've looked through the commit history, and I think i was confusing this with some other fixes that I made to the readiness logic. I don't see this affecting it so if tests passes, +1.

Reply:
@lulf  I  don't see a problem, but I don't find it particularly easy code to work with.  What is it that mutates the address record?   I don't think its this.update_readiness() and it maintains a separate structure (readiness - which is used only by brokered).

The cowardly approach would be to repeat the filter, which I can do if preferred.